pkg/optimizely/cache.go

```diff
@@ -226,7 +249,7 @@ func defaultLoader(
 		}
 
 		var clientUserProfileService decision.UserProfileService
-		if clientUserProfileService = getUserProfileService(sdkKey, userProfileServiceMap, conf); clientUserProfileService != nil {
+		if clientUserProfileService = getUserProfileService(ctx, sdkKey, userProfileServiceMap, conf); clientUserProfileService != nil {
```

Comment:
Why we need to change this method.

Reply:
Earlier, We were giving `UPS` a different context altogether which might've resulted in `UPS` requests still running for a little while after agent had shut down. This fix is to pass the original agent context to `UPS` so that once agents shuts down, `UPS` terminates all its requests as well.